apps/genetics/src/utils/variant.ts

```diff
@@ -57,14 +66,18 @@ export const variantPopulations = [
   { code: "OTH", description: "Other (population not assigned)" },
 ];
 
-export function variantParseGenesForVariantSchema(data) {
-  const genesForVariantSchema = data.genesForVariantSchema;
+type VariantSchemaData = {
+  genesForVariantSchema?: G2VSchema;
+};
+export function variantParseGenesForVariantSchema(data?: VariantSchemaData) {
+  const genesForVariantSchema = data?.genesForVariantSchema;
+  if (!genesForVariantSchema) return;
   const currentQtls = genesForVariantSchema.qtls;
   if (currentQtls.length === 0) return genesForVariantSchema;
   //possibly add new fields
   if (currentQtls[0].id === "pqtl") {
     const pqtl = currentQtls[0];
-    const restQtls = currentQtls.slice(1, pqtl.length);
```

Comment:
I think this was a bug, where `pqtl.length` was supposed to be `currentQtls.length`? I don't believe `pqtl` is an array to my knowledge. Either way, removing the parameter should be fine since no `end` param means to go til the end of the array which is what `restQtls` is supposed to be I think

Reply:
Looking at the code, I do agree with your assessment I think it was a bug